tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -105,7 +105,10 @@ def resolve_ips(self):
             sockaddr[0] for (_, _, _, _, sockaddr) in socket.getaddrinfo(self.host, self.port, 0, 0, socket.IPPROTO_TCP)
         ]
         if not self.multiple_ips:
-            self._addrs = self._addrs[:1]
+            if not is_ipv6(self.host):
+                self._addrs = [socket.gethostbyname(self.host)]
+            else:
+                self._addrs = self._addrs[:1]
```

Comment:
How about something like this, keeping the same approach:

```python
    def resolve_ips(self):
        if self.socket_type == socket.AF_INET:
            # gethostbyname_ex only translates to IPv4 addresses
            _, _, self._addrs = socket.gethostbyname_ex(self.host)
        else:
            self._addrs = [
                sockaddr[0] for (_, _, _, _, sockaddr) in socket.getaddrinfo(self.host, self.port, 0, 0, socket.IPPROTO_TCP)
            ]
        if not self.multiple_ips:
                self._addrs = self._addrs[:1]
```

- Using `gethostbyname_ex` instead of `gethostbyname` to support multiple ips so that we don't keep having a case where we would still fail (the code as it is would still fail if `multiple_ips` was set to `true`). This would also keep the logic more homogeneous. You would have to update the test mocking, of course.
- Comparing against the value of `socket_type` instead of going through the `is_ipv6` function again might make more sense.
- Adding a comment on gethostbyname(_ex) so that the reader doesn't need to go look up the documentation to know why it's being used.

Reply:
Thanks, those are good points! I'll make those changes

Reply:
Hey @alopezz, `gethosbyname_ex` worked, but we still need to use the actual `socket_type` to prevent the `[Errno -9] Address family for hostname not supported` error. I ended up comparing against the `socket_type` returned from `getaddrinfo` so `gethostbyname_ex` is no longer needed.